packages/sd-jwt-vc/src/sd-jwt-vc-instance.ts

```diff
@@ -113,12 +123,167 @@ export class SDJwtVcInstance extends SDJwtInstance<SdJwtVcPayload> {
+        arrayBuffer,
+        alg,
+      );
+      const integrityHash = integrity.split('-')[1];
+      const hash = Array.from(new Uint8Array(hashBuffer))
+        .map((byte) => byte.toString(16).padStart(2, '0'))
+        .join('');
+      if (hash !== integrityHash) {
+        throw new Error(
+          `Integrity check for ${url} failed: is ${hash}, but expected ${integrityHash}`,
+        );
+      }
+    }
+  }
+
+  /**
+   * Fetches the content from the url with a timeout of 10 seconds.
+   * @param url
+   * @returns
+   */
+  private async fetch<T>(url: string, integrity?: string) {
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), 10000);
+    try {
+      const response = await fetch(url, {
+        signal: controller.signal,
+      });
+      if (!response.ok) {
+        throw new Error(await response.text());
+      }
+      await this.validateIntegrity(response.clone(), url, integrity);
+      return response.json() as Promise<T>;
+    } finally {
+      clearTimeout(timeoutId);
+    }
+  }
+
+  /**
+   * Loads the schema either from the object or as fallback from the uri.
+   * @param typeMetadataFormat
+   * @returns
+   */
+  private async loadSchema(typeMetadataFormat: TypeMetadataFormat) {
+    //if schema is present, return it
+    if (typeMetadataFormat.schema) return typeMetadataFormat.schema;
+    if (typeMetadataFormat.schema_uri) {
+      const schema = await this.fetch<SchemaObject>(
+        typeMetadataFormat.schema_uri,
+        typeMetadataFormat['schema_uri#Integrity'],
+      );
+      return schema;
+    }
+    throw new Error('No schema or schema_uri found');
+  }
+
+  /**
+   * Verifies the VCT of the SD-JWT-VC. Returns the type metadata format. If the schema does not match, an error is thrown. If it matches, it will return the type metadata format.
+   * @param result
+   * @returns
+   */
+  private async verifyVct(
+    result: VerificationResult,
+  ): Promise<TypeMetadataFormat | undefined> {
+    const fetcher: VcTFetcher =
+      this.userConfig.vctFetcher ?? this.vctFetcher.bind(this);
+    const typeMetadataFormat = await fetcher(
+      result.payload.vct,
+      result.payload['vct#Integrity'],
+    );
+
+    if (typeMetadataFormat.extends) {
+      // implement based on https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-04.html#name-extending-type-metadata
+      //TODO: needs to be implemented. Unclear at this point which values will overwrite the values from the extended type metadata format
+    }
+
+    //init the json schema validator, load referenced schemas on demand
+    const schema = await this.loadSchema(typeMetadataFormat);
+    const loadedSchemas = new Set<string>();
+    // init the json schema validator
+    const ajv = new Ajv({
+      loadSchema: async (uri: string) => {
+        if (loadedSchemas.has(uri)) {
+          return {};
+        }
+        const response = await fetch(uri);
```

Comment:
```suggestion
        const response = await this.fetch(uri);
```
Should it be `this.fetch`?

Reply:
It does not really matter. The internal fetch function has an integrity validation. This is not passed when loading schemas. However I will check if schemas have an integrity validation and if we can add it then like we are doing it with the other resources from vct

Reply:
@lukasjhan I found no standard for validating the integrity of a referenced $schema value. At this point I do not see any privacy concerns when we are not validating it for now. In case we see something, we can open it.